larpandora/LArPandoraInterface/LArPandoraInput.cxx

```diff
@@ -215,17 +215,53 @@ void LArPandoraInput::CreatePandoraLArTPCs(const Settings &settings, const LArDr
 
 void LArPandoraInput::CreatePandoraDetectorGaps(const Settings &settings, const LArDriftVolumeList &driftVolumeList, const LArDetectorGapList &listOfGaps)
 {
+    //ATTN - Unlike SP, DP detector gaps are not in the drift direction 
+    art::ServiceHandle<geo::Geometry const> theGeometry;
+    const bool isDualPhase(theGeometry->MaxPlanes() == 2);
+    
     mf::LogDebug("LArPandora") << " *** LArPandoraInput::CreatePandoraDetectorGaps(...) *** " << std::endl;
-
+    
     if (!settings.m_pPrimaryPandora)
         throw cet::exception("LArPandora") << "CreatePandoraDetectorGaps - primary Pandora instance does not exist ";
-
+    
     const pandora::Pandora *pPandora(settings.m_pPrimaryPandora);
-
+    
     for (const LArDetectorGap &gap : listOfGaps)
     {
         PandoraApi::Geometry::LineGap::Parameters parameters;
-
+        
+        if (isDualPhase)
+        {
+            const float gapSizeY (std::fabs(gap.GetY2() - gap.GetY1())); //Could have chosen Z here, resulting in switching Y<->Z and U<->V in the try{...} block below
+            
+            const float maxGapSize (30.); //Match the maxDisplacement variable hard-coded value in LArPandoraGeometry
+             
+            try
+            {
+                parameters.m_lineGapType = ((gapSizeY > maxGapSize) ? pandora::TPC_WIRE_GAP_VIEW_U : pandora::TPC_WIRE_GAP_VIEW_V); //If gapSizeY is too large then the gap is in Z, therefore should be in kU (i.e. kZ)
+                parameters.m_lineStartX = gap.GetX2();
+                parameters.m_lineEndX = gap.GetX1();
+                parameters.m_lineStartZ = ((gapSizeY > maxGapSize) ? gap.GetZ1() : gap.GetY1());
+                parameters.m_lineEndZ = ((gapSizeY > maxGapSize) ? gap.GetZ2() : gap.GetY2());
+            }
+            
```

Comment:
Suggest remove this empty line, to attach this try/catch pair, then add a line below for the next try/catch pair.